File: frontend/src/app/models/paciente.model.ts
import { Usuario } from './usuario.model';

export class Paciente extends Usuario {
  constructor(
    id: number,
    nombre: string,
    apellidos: string,
    usuario: string,
    clave: string,
    public nss: string,
    public numTarjeta: string,
    public telefono: string,
    public direccion: string,
    public medicoIds: number[] = [],
  ) {
    super(id, nombre, apellidos, usuario, clave);
  }
}

File: backend/CitasMedicas.Application/Models/PacienteModel.cs
namespace CitasMedicas.Application.Models;

public class PacienteModel
{
    public int Id { get; set; }

    public string Nombre { get; set; } = string.Empty;

    public string Apellidos { get; set; } = string.Empty;

    public string Usuario { get; set; } = string.Empty;

    public string? Clave { get; set; }

    public string NSS { get; set; } = string.Empty;

    public string NumTarjeta { get; set; } = string.Empty;

    public string Telefono { get; set; } = string.Empty;

    public string Direccion { get; set; } = string.Empty;

    public List<int> MedicoIds { get; set; } = [];
}